### ISSUE TITLE
Add hooks for PyGTK themes

### ISSUE THREAD

**Issue by pyinstaller-tickets-migration** [pyinstaller/pyinstaller#969]
**Original date**: 2007/06/20
**Original reporter**: _anonymous_

I have tried to build a simple PyGTK program, by using PyInstaller 1.3, accordingly to the suggestions reported in the tickets about PyGTK (hidden imports and Pango files).

Under Windows XP, I have chosen a "theme" for GTK applications, by using the "Theme Selector" application; theme is a sort of visual "look and feel": when I run the script, the chosen theme applies to the window controls; when I run the executable, a standard theme is used instead.

It is probably due to some GTK library settings: if you have fixed the pango problem, I think you should also be able to fix this. All I can tell you is that custom themes (for example, the ones in the user's computer - since it's the user who chooses the themes) are kept under $GTK_HOME/share/themes, and every theme has got a separate directory.

I do not know if this applies to Linux, too.

**Comment by htgoebel** [comment]
Pushing forward to next milestone.

**Comment by yellowtailfan** [comment]
To get the MS-Windows theme to work from inside a pyinstaller bundle, I had to also set the GTK_EXE_PREFIX environment variable inside the script to the value of sys._MEIPASS (after first detecting if the app is running in frozen mode and before the first gtk.Window call). This is in addition to the .spec file changes explained by MeV above.

Until I did that, I was getting this error:
```
Gtk-WARNING: Unable to locate theme engine in module_path: “wimp”
```
I'm running pyinstaller 3.3.1 from within MSYS2 MinGW with PyGTK 2.24.32.

I was able to diagnose the problem by printing the value of gtk.rc_get_module_dir() from inside the frozen script, and seeing that the path returned was incorrect.
